docs/framework/network-programming/using-udp-services.md

```diff
@@ -21,156 +21,141 @@ author: "mcleblanc"
+
+To send a datagram using UDP, you must know the network address of the network device hosting the service you need and the UDP port number that the service uses to communicate. The Internet Assigned Numbers Authority (Iana) defines port numbers for common services (see [Service Name and Transport Protocol Port Number Registry](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml)). Services not on the Iana list can have port numbers in the range 1,024 to 65,535.
+
+Special network addresses are used to support UDP broadcast messages on IP-based networks. The following discussion uses the IP version 4 address family used on the Internet as an example.
+
+IP version 4 addresses use 32 bits to specify a network address. For class C addresses using a netmask of 255.255.255.0, these bits are separated into four octets. When expressed in decimal, the four octets form the familiar dotted-quad notation, such as 192.168.100.2. The first two octets (192.168 in this example) form the network number, the third octet (100) defines the subnet, and the final octet (2) is the host identifier.
+
+Setting all the bits of an IP address to one, or 255.255.255.255, forms the limited broadcast address. Sending a UDP datagram to this address delivers the message to any host on the local network segment. Because routers never forward messages sent to this address, only hosts on the network segment receive the broadcast message.
+
+Broadcasts can be directed to specific portions of a network by setting all bits of the host identifier. For example, to send a broadcast to all hosts on the network identified by IP addresses starting with 192.168.1, use the address 192.168.1.255.
+
+The following code example uses a <xref:System.Net.Sockets.UdpClient> to listen for UDP datagrams on port 11,000. The client receives a message string and writes the message to the console.
+
+```vb
+Imports System
+Imports System.Net
+Imports System.Net.Sockets
+Imports System.Text
+
+Public Class UDPListener
+   Private Const listenPort As Integer = 11000
+   
+   Private Shared Sub StartListener()
+      Dim listener As New UdpClient(listenPort)
+      Dim groupEP As New IPEndPoint(IPAddress.Any, listenPort)
+      Try
+         While True
+            Console.WriteLine("Waiting for broadcast")
+            Dim bytes As Byte() = listener.Receive(groupEP)
+            Console.WriteLine("Received broadcast from {0} :", groupEP)
+            Console.WriteLine(Encoding.ASCII.GetString(bytes, 0, bytes.Length))
+         End While
+      Catch e As Exception
```

Comment:
We're trying to eliminate handling Exception. [UdpClient.Receive](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.udpclient.receive?view=netframework-4.7.2) throws a SocketException if an error occurs when accessing the socket. Can you handle that instead of Exception?